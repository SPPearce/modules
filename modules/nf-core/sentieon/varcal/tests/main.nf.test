nextflow_process {

    name "Test Process SENTIEON_VARCAL"
    config "./nextflow.config"
    script "../main.nf"
    process "SENTIEON_VARCAL"
   
    tag "modules"
    tag "modules_nfcore"
    tag "sentieon"
    tag "sentieon/varcal"
   
    
    
    test("test_sentieon_varcal") {
        
        when {
            params {
                module_args = '--var_type SNP --annotation QD --annotation MQ --annotation FS --annotation SOR'
            }
            process {
                """
                input[0] = [[id:'test'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz.tbi',checkIfExists:true)
                    ]
                input[1] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/hapmap_3.3.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_omni2.5.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_phase1.snps.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/dbsnp_138.hg38.vcf.gz',checkIfExists:true)
                    ]
                input[2] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/hapmap_3.3.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_omni2.5.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_phase1.snps.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/dbsnp_138.hg38.vcf.gz.tbi',checkIfExists:true)
                    ]
                input[3] = [
                    '--resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.hg38.vcf.gz',
                    '--resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.hg38.vcf.gz',
                    '--resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.hg38.vcf.gz',
                    '--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp_138.hg38.vcf.gz'
                    ]
                input[4] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                input[5] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
    
    
    test("test_sentieon_varcal -- stub") {
        options '-stub'
        when {
            params {
                module_args = '--var_type SNP --annotation QD --annotation MQ --annotation FS --annotation SOR'
            }
            process {
                """
                input[0] = [[id:'test'],// meta map
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/gatk/haplotypecaller_calls/test2_haplotc.ann.vcf.gz.tbi',checkIfExists:true)
                    ]
                input[1] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/hapmap_3.3.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_omni2.5.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_phase1.snps.hg38.vcf.gz',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/dbsnp_138.hg38.vcf.gz',checkIfExists:true)
                    ]
                input[2] = [
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/hapmap_3.3.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_omni2.5.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/1000G_phase1.snps.hg38.vcf.gz.tbi',checkIfExists:true),
                    file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/germlineresources/dbsnp_138.hg38.vcf.gz.tbi',checkIfExists:true)
                    ]
                input[3] = [
                    '--resource:hapmap,known=false,training=true,truth=true,prior=15.0 hapmap_3.3.hg38.vcf.gz',
                    '--resource:omni,known=false,training=true,truth=false,prior=12.0 1000G_omni2.5.hg38.vcf.gz',
                    '--resource:1000G,known=false,training=true,truth=false,prior=10.0 1000G_phase1.snps.hg38.vcf.gz',
                    '--resource:dbsnp,known=true,training=false,truth=false,prior=2.0 dbsnp_138.hg38.vcf.gz'
                    ]
                input[4] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta',checkIfExists:true)
                input[5] = file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/chr21/sequence/genome.fasta.fai',checkIfExists:true)
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}