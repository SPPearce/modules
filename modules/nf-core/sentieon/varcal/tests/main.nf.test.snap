{
    "homo sapiens -- stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idx:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tranches:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "3": [
                    [
                        {
                            "id": "test"
                        },
                        "testplots.R:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "4": [
                    "versions.yml:md5,12f6cca4dd81f3db921bcd06145f62e8"
                ],
                "idx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.idx:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "plots": [
                    [
                        {
                            "id": "test"
                        },
                        "testplots.R:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "recal": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "tranches": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tranches:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,12f6cca4dd81f3db921bcd06145f62e8"
                ]
            },
            {
                "SENTIEON_VARCAL": {
                    "sentieon": 202308.03
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-21T13:06:16.191785081"
    },
    "homo sapiens": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal:md5,9003bdfdd46ddf5d24d407de603fa254"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal.idx:md5,e1fbfed866e80f863a5c212c7bef8da2"
                    ]
                ],
                "2": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tranches:md5,575517ff6abbc8d7bf7aee4eef449d29"
                    ]
                ],
                "3": [
                    
                ],
                "4": [
                    "versions.yml:md5,12f6cca4dd81f3db921bcd06145f62e8"
                ],
                "idx": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal.idx:md5,e1fbfed866e80f863a5c212c7bef8da2"
                    ]
                ],
                "plots": [
                    
                ],
                "recal": [
                    [
                        {
                            "id": "test"
                        },
                        "test.recal:md5,9003bdfdd46ddf5d24d407de603fa254"
                    ]
                ],
                "tranches": [
                    [
                        {
                            "id": "test"
                        },
                        "test.tranches:md5,575517ff6abbc8d7bf7aee4eef449d29"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,12f6cca4dd81f3db921bcd06145f62e8"
                ]
            },
            {
                "SENTIEON_VARCAL": {
                    "sentieon": 202308.03
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-04-21T13:05:48.378792551"
    }
}